File: ocaml/day05.ml
let instructions = Intcode.read_instructions "inputs/05.txt"

let solve ~part instructions =
  let input = if part = 1 then 1 else 5 in
  let ram_size = List.length instructions in
  instructions
  |> Intcode.initialize_computer ~ram_size
  |> Intcode.receive input
  |> Intcode.run_until_halt
  |> Intcode.get_last_output
  |> Printf.printf "%d\n"

let () =
  instructions |> solve ~part:1;
  instructions |> solve ~part:2

File: python/day05.py
from intcode import *


def solve(instr, part):
    input_instruction = 1 if part == 1 else 5
    comp = Computer(instr, len(instr))
    comp.receive(input_instruction)
    comp.run()
    return comp.out_queue[-1]


instructions = read_intcode_input(5)

print(solve(instructions, 1))
print(solve(instructions, 2))
